expression/builtin_control.go

```diff
@@ -54,6 +54,14 @@ var (
 	_ builtinFunc = &builtinIfJSONSig{}
 )
 
+func maxlen(lhsFlen, rhsFlen int) int {
+	// -1 indicates that the length is unknown, such as the case for expressions.
+	if lhsFlen < 0 || rhsFlen < 0 {
+		return 21
```

Comment:
Why 21 here?

Reply:
This was the value observed in MySQL when the length is unknown.

Reply:
I found a constant that matches this (MaxRealWidth). In TiDB it is 23, but I think that's okay.